src/forms/form-fields/form-template.tsx

```diff
@@ -0,0 +1,147 @@
+  isLoading: boolean;
+}
+
+interface ModalFormTemplateProps {
+  schema: Schema;
+  submitLabel?: ReactNode;
+  formFields: ReactNode[];
+  modalProps: Partial<InternalModalProps>;
+}
+
+const ModalFormTemplate: React.ComponentType<ModalFormTemplateProps> = ({
+  schema,
+  formFields,
+  submitLabel,
+  modalProps: { isLoading, title, description, onClose, variant }
+}) => {
+  const { handleSubmit } = useFormApi();
+  const formatMessage = useFormatMessage();
+  return ReactDOM.createPortal(
+    <div>
+      <Backdrop>
+        <FocusTrap className="pf-l-bullseye">
+          <StyledForm
+            onSubmit={isLoading ? () => undefined : handleSubmit}
+            variant={variant}
+            className="pf-c-modal-box pf-m-sm"
+          >
+            <ModalBoxCloseButton onClose={onClose} />
+            <ModalBoxHeader>
+              <h1 id="modal-form-title" className="pf-c-modal-box__title">
+                {title}
+              </h1>
+              {description && (
+                <div
+                  id="modal-form-description"
+                  className="pf-c-modal-box__description"
+                >
+                  {description}
+                </div>
+              )}
+            </ModalBoxHeader>
+            <ModalBoxBody>
+              <div>
+                <div className="pf-c-form">
+                  {isLoading ? (
+                    <SpinnerWrapper className="pf-u-m-md">
+                      <Spinner />
+                    </SpinnerWrapper>
+                  ) : (
+                    <Fragment>
+                      {schema.title}
+                      {formFields}
+                    </Fragment>
+                  )}
+                </div>
+              </div>
+            </ModalBoxBody>
+            <ModalBoxFooter>
+              <FormSpy
+                subscription={{
+                  invalid: true,
+                  validating: true,
+                  pristine: true,
+                  submitting: true
+                }}
+              >
+                {({ pristine, invalid, validating, submitting }) => (
+                  <Fragment>
+                    <Button
+                      type="submit"
+                      isDisabled={
+                        pristine || invalid || validating || submitting
+                      }
+                    >
+                      {submitLabel || formatMessage(labelMessages.save)}
+                    </Button>
+                    <Button variant="link" onClick={onClose}>
+                      {formatMessage(labelMessages.cancel)}
+                    </Button>
+                  </Fragment>
+                )}
+              </FormSpy>
+            </ModalBoxFooter>
+          </StyledForm>
+        </FocusTrap>
+      </Backdrop>
+    </div>,
+    document.body
+  );
+};
+
+export interface FormTemplateProps extends Pf4FormTemplateProps {
+  formContainer: 'default' | 'modal';
```

Comment:
Maybe we should change this to `isModal` to make the code a little bit simpler (if you don't expect more variants to come)

Reply:
We could do that. :+1: